### ISSUE TITLE
discussion: why pauls-rpc vs. patching up electrons existing rpc?

### ISSUE THREAD

**Issue by cdaringe** [pfrazee/pauls-electron-rpc#2]
hey @pfrazee, nice to meet you the other night at at pdxnode (i was shaved head guy with lots of Qs).

i stumbled across this pkg browsing the beaker src.

i see there are some extra _things™_ that this rpc package handles vs electron's builtin (timeouts, streams, etc).  my gut feeling is that it would be great to be able to just use their rpc.  the `remote` api is much more concise, albeit less capable.  [here's what their rpc server supports now](https://github.com/electron/electron/blob/master/lib/browser/rpc-server.js#L58-L82).  i'm wondering if you see that as a feasiblity if some of the lacking features were ported in over there, or if it's even desirable from your perspective.  thx!

**Comment by pfrazee** [comment]
Yeah it might be doable to switch over. My reason was, this is a pretty critical piece of code (for security, performance, *and* correctness) and I wasn't sure I could rely on `remote` for everything. I figured it was better to have my own RPC which I could control.

One feature which I need, and they don't have (AFAICT) is knowledge of the sender. I use that to enforce permissions.
